docs/containers-storage-architecture.md

```diff
@@ -0,0 +1,76 @@
+## containers-storage 1 "May 2024"
+
+## NAME
+containers-storage-architecture - The implementation of containers-storage 
+
+NOTE: These are implementation details subject to change. This document
+is intended to be used only as an aid for debugging and development
+of the containers-storage system. Do *not* directly access the filesystem
+components below from external code, only from the Go API.
+
+## DESCRIPTION
+The *containers/storage* library manages the following object types:
+
+- layer: A filesystem tree, along with metadata
+- image: A reference to a layer, along with metadata
+- container: A layer whose parent is an image
+
+## Layer storage
+
+### `overlay/`
+
+The `overlay/` directory contains a set of layers, named
+by their digest (note, this is the "diffid" or *uncompressed*
+digest, not the compressed one).
```

Comment:
No, this is a “chain ID” identifying the precise sequence of parent1→parent2→…→child layer IDs; and with zstd:chunked, also identifying the contents of the layer depending on how we actually created it.

I.e. a layer with the same DiffID can have _several_ physical objects in this directory.